### PR TITLE
Add BLE device onboarding

### PR DESCRIPTION
**Pull request by ChewingGlass**

### CODE REVIEW

File: src/features/hotspot-onboarding/iot-ble/AddGatewayBle.tsx [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+      )
+      return
+    }
+
+    const onboardAddress = await getOnboardingAddress()
+    const onboardRecord = await getOnboardingRecord(onboardAddress)
+
+    if (!onboardRecord) {
+      throw new Error(
+        `This hotspot does not exist in the onboarding server. Contact your manufacturer to have them approve hotspot with id ${onboardAddress}`,
```
Comment (matthewcarlreetz) [inline src/features/hotspot-onboarding/iot-ble/AddGatewayBle.tsx:68]: should we localize the strings in this file?

File: src/features/hotspot-onboarding/iot-ble/AddGatewayBle.tsx [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    const makerSolBalance = (
+      await anchorProvider.connection.getAccountInfo(
+        new PublicKey(makerSolAddr),
+      )
+    )?.lamports
+    if (
+      !makerSolBalance ||
+      makerSolBalance / LAMPORTS_PER_SOL < 0.00089088 + 0.00001
+    ) {
+      throw new Error(
```
Comment (matthewcarlreetz) [inline src/features/hotspot-onboarding/iot-ble/AddGatewayBle.tsx:85]: Is there a scenario where the user would pay onboarding fees?

File: src/features/hotspot-onboarding/iot-ble/AddGatewayBle.tsx [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+      ownerAddress: accountAddress,
+      payerAddress: onboardRecord?.maker.address,
+    })
+    const tx = AddGatewayV1.fromString(txnStr)
+    if (!tx?.gateway?.b58) {
+      throw new Error('Error signing gateway txn')
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    function wrapProgramError(e: any) {
```
Comment (matthewcarlreetz) [inline src/features/hotspot-onboarding/iot-ble/AddGatewayBle.tsx:100]: I feel like we should put this insufficient maker funds stuff somewhere that we can reuse it

File: src/features/hotspot-onboarding/iot-ble/AddGatewayBle.tsx [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+          `Manufacturer ${onboardRecord?.maker.name} does not have enough SOL or Data Credits to onboard this hotspot. Please contact the manufacturer to resolve this issue.`,
+        )
+      }
+      if (e.InstructionError) {
+        throw new Error(`Program Error: ${JSON.stringify(e)}`)
+      }
+      throw e
+    }
+
+    const createTxns = await onboardingClient.createHotspot({
```
Comment (matthewcarlreetz) [inline src/features/hotspot-onboarding/iot-ble/AddGatewayBle.tsx:116]: could we add maker insufficient funds checks to the onboarding server create and update functions?

File: src/utils/walletApiV2.ts [hunk 5]
```diff
@@ -131,6 +132,15 @@ export const getRecommendedDapps = async () => {
   return data
 }
 
+export const getOnboardingDevices = async () => {
+  const { data, ...rest } = await axiosInstance.get<OnboardableDevice[]>(
+    '/onboardingDevices',
+  )
+  console.log("rest", rest)
```
Comment (matthewcarlreetz) [inline src/utils/walletApiV2.ts:139]: remove

File: src/features/hotspot-onboarding/iot-ble/AddGatewayBle.tsx (hunk 2 above)
Comment (ChewingGlass) [reply src/features/hotspot-onboarding/iot-ble/AddGatewayBle.tsx:85]: Not yet. There might be if the defunct maker hip passes

File: src/features/hotspot-onboarding/iot-ble/AddGatewayBle.tsx (hunk 3 above)
Comment (ChewingGlass) [reply src/features/hotspot-onboarding/iot-ble/AddGatewayBle.tsx:100]: The problem is it's wrapping a very generic transaction error, and the message wouldn't be correct in all (most) cases. It's pretty specific to these txns.

File: src/features/hotspot-onboarding/iot-ble/AddGatewayBle.tsx (hunk 4 above)
Comment (ChewingGlass) [reply src/features/hotspot-onboarding/iot-ble/AddGatewayBle.tsx:116]: We could, but it wont be reliable as these transactions could be formed when there's sufficient sol, but then run when someone else took that sol or dc.

**Review by bryzettler — APPROVED** [review]
Build failing and packages should be updated to 0.4.0 but looks good to me.

**Pull request merged**